python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -637,3 +642,23 @@ def _validate_dagster_components_availability(context: DgContext) -> None:
             )
     elif not context.has_executable("dagster-components"):
         exit_with_error(generate_missing_dagster_components_error_message())
+
+
+def _validate_project_venv_activated(context: DgContext) -> None:
+    if not context.config.project:
+        raise DgError(
+            "`_validate_project_venv_activated` is only available in a Dagster project context"
+        )
+    activated_venv = get_activated_venv()
+    project_venv = context.root_path / ".venv"
+    if (
+        context.config.project.python_environment.active
```

Comment:
I do wish this key was a little more self-descriptive, something like `use_active`, `use_activated`, `from_activated`, `use_system` etc. This reads at first glance as whether an environment _is_ active rather than specifying that the environment is sourced from the system activated environment. I also find this a bit hard to parse meaning from in the CLI arg case of `--python-environment active`

Reply:
Yeah @benpankow a good point. Counterpoint is that this is the language that uv uses, and I think that will be the standard going forward.

e.g.

`uv run -h  | grep active` 

`--active           Prefer the active virtual environment over the project's virtual environment`

Reply:
yea naming here is tough. `active` was chosen because it matches `uv`:

```
     --active                                 Prefer the active virtual environment over the project's virtual environment
```

Reply:
jinx again